src/mappings/event/loans.ts

```diff
@@ -566,11 +566,12 @@ export async function liquidateLoan(
         ctx.log.warn(`UNKOWN EVENT VERSION: LoansLiquidatedBorrowEvent`);
         return;
     }
-
+    
     const amountRepaidExchangeRate = await getExchangeRate(ctx, block.timestamp, amountRepaidToken, Number(amountRepaid));
     const seizedCollateralExchangeRate = await getExchangeRate(ctx, block.timestamp, seizedCollateralToken, Number(seizedCollateral));
     const liquidationCostBtc = seizedCollateralExchangeRate.btc.toNumber() - amountRepaidExchangeRate.btc.toNumber();
-    const liquidationCostUsdt = seizedCollateralExchangeRate.usdt.toNumber() - amountRepaidExchangeRate.usdt.toNumber();
+
+    liquidationCost = BigInt(liquidationCostBtc * seizedCollateralExchangeRate.btcExchangeRate.toNumber());
```

Comment:
We are risking losing precision in this multiplication since the actual calculation is still of type `number` before being turned into a `BigInt`.
(To be fair: the other pre-existing operations have similar issues, but multiplications tend to get above the max number faster)

So, I'd probably lean on using the `Big` in `seizedCollateralExchangeRate.btcExchangeRate` and do something like:
```
const liquidationCost = BigInt(seizedCollateralExchangeRate.btcExchangeRate.mul(liquidationCostBtc.toString()).toString());
```

Using `.toString()` preserves the full number whereas `.toNumber()` might not be able to do that. And both `BigInt()` as well as `Big()` constructors can take a String.
(Side note: I also added `const` because we don't need to declare it as `let liquidationCost` above if we only set it there.)

Reply:
This awkwardness highlights one of the reasons why I recommended using `ExchangeRate` from the monetary lib in the first place. It makes working with these values more consistent.

But that type of change may well be better done as a future refactoring issue.

Reply:
Ah, I am really starting to see the value of using ExchangeRate. It was just much easier to use what the grants had alread implementated and keep this file consistant. I will open a github issue to refactor what the grants team added using the monetary lib.